recipes/repository.rb

```diff
@@ -37,6 +44,30 @@
 when 'rhel', 'fedora'
   include_recipe 'yum'
 
+  # Import new RPM key
+  if node['datadog']['yumrepo_gpgkey_new']
+    # gnupg is required to check the downloaded key's fingerprint
+    package 'gnupg' do
+      action :install
+    end
+
+    # Download new RPM key
```

Comment:
I don't think you need to download the file... You should be able to `rpm --import "#{yum_protocol}://yum.datadoghq.com/DATADOG_RPM_KEY.public.E09422B3"`